docs/_posts/examples/3400-01-14-filter-features-within-map-view.html

```diff
@@ -150,30 +156,33 @@
         "layout": {
             "icon-image": "airport-15",
             "icon-padding": 0
-        },
-        "filter": ["in", "abbrev", ""]
+        }
     });
-
     map.on('moveend', function() {
-        var features = map.querySourceFeatures('airports', {
-            sourceLayer: 'ne_10m_airports'
-        });
+        var features = map.queryRenderedFeatures({layers:['airport']});
 
         if (features) {
+            var renderedAbbrevs = {};
+            // It is possible for the same feature to be included twice in the output of
+            // `map.queryRenderedFeatures()` if features span tile boundaries, so we
+            // filter them out by a unique key
```

Comment:
A feature may be included more than twice in the output. Non-point features may be included an unbounded number of times in the output. 

The language in the [docs](https://www.mapbox.com/mapbox-gl-js/api/#Map#queryRenderedFeatures) is good. Feel free to borrow from there.